cns/ipampoolmonitor/ipampoolmonitor.go

```diff
@@ -249,3 +248,14 @@ func (pm *CNSIPAMPoolMonitor) Update(scalar nnc.Scaler, spec nnc.NodeNetworkConf
 
 	return nil
 }
+
+//this function sets the values for state in IPAMPoolMonitor Struct
+func (pm *CNSIPAMPoolMonitor) GetStateSnapshot() cns.IpamPoolMonitorStateSnapshot{
+	defer pm.mu.Unlock()
+	pm.mu.Lock()
```

Comment:
This works as intended because scope is limited, and I can see its the style in the rest of the file, but for future reference, it is not a good practice to defer a closing/unlocking action before a lock has been taken/something has been opened.

Reply:
ack, this makes sense, I can put the defer after the lock is called.